bika/health/upgrade/v01_01_003.py

```diff
@@ -31,7 +35,77 @@ def upgrade(tool):
     # Ensure health's skins have always priority over core's
     setup = portal.portal_setup
     setup.runImportStepFromProfile(profile, "skins")
+    setup.runImportStepFromProfile(profile, 'workflow')
+    setup.runImportStepFromProfile(profile, 'propertiestool')
+
+    add_doctor_action_for_client(portal)
+    ut.addIndexAndColumn('portal_catalog', 'allowedRolesAndUsers', 'FieldIndex')
+    ut.addIndex('portal_catalog', 'getPrimaryReferrerUID', 'FieldIndex')
+    ut.refreshCatalogs()
```

Comment:
This will try to reindex these two fields for the whole catalog (which is "portal_catalog"!), so I expect this will take loong. Since you only added the field in Doctor content type, I feel a better approach would be to only reindex Doctor objects individually instead of the whole catalog. And this is not even necessary, cause you are already doing this reindexing stuff in `update_permissions_clients`. In summary, I'd remove this `ut.refreshCatalogs()` from here and maybe add these two indexes (and column) manually inside `update_permissions_clients` function.